### PR TITLE
Wedge slicer minor upgrages

### PR DESCRIPTION
**Pull request by ehewins**
## Description

These are just some minor readability improvements and optimisations, based on the suggestions @rozyczko made on the 2550-wedge-slicer Pull Request which was merged during the meeting earlier today. Once this is merged, I think we can delete the branch.

## How Has This Been Tested?

The same tests as were run on the version of the branch merged earlier today. Everything still functions as normal.

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [x] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [X] User documentation is available and complete (if required)
- [X] Developers documentation is available and complete (if required)
- [X] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

**Review by rozyczko — APPROVED** [review]
This looks good!

**Review by krzywon — APPROVED** [review]
I have one minor suggestion, but a good cleanup of the code overall. I tested this using the branch from sasview/sasdata#41 and the wedge slicers, in both Q and phi, work.

File: src/sas/qtgui/Plotting/Slicers/ArcInteractor.py [hunk 1]
```diff
@@ -73,27 +66,23 @@ def update(self, theta=None, phi=None, r=None):
         :param phi: angle from the centre point on the arc to each of its edges
```
Comment (astellhorn) [inline src/sas/qtgui/Plotting/Slicers/ArcInteractor.py:66]: Why is it called "theta" for the angular value of the middle of the arc, whereas in the GUI its "phi", and here "phi" being the parameter that is called "delta_phi" in the GUI? That is very misleading.

File: src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py [hunk 2]
```diff
@@ -211,38 +211,41 @@ def _post_data(self, new_sector=None, nbins=None):
...
-                print("Number of bins cannot be <= 0. Please adjust.")
-                isValid = False
-        return isValid
+                return "Number of bins cannot be <= 0. Please adjust."
+            return None
+
+        validators = {
+            'r_min': lambda value: check_radius_difference('r_min', 'r_max', value),
+            'r_max': lambda value: check_radius_difference('r_max', 'r_min', value),
+            'delta_phi [deg]': check_phi_difference,
```
Comment (astellhorn) [inline src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py:238]: setting delta_phi very low in windows gui (e.g. 0.000001 deg) does not give any error message in the log, but still the plot window does not react, so an error message should be created!

**Comment by astellhorn** [comment]
General comments on the wedge slicers (Windows GUI):
1) Regarding the x-axis range: 
-->The phi slicer has wrong x-axis labels (might there be a pi too much? Its 3.14 "deg" where it should be 0deg
image

2) Regarding the x-axis fonds:
-->x-Axis fonds are overlapping for both Q and phi slicers (depending on the axis range), whereas for other kind of slicers there is no problem.

image

File: src/sas/qtgui/Plotting/Slicers/ArcInteractor.py (hunk 1 above)
Comment (ehewins) [reply src/sas/qtgui/Plotting/Slicers/ArcInteractor.py:66]: Here I was copying how the variables are named in SectorSlicer, which has exactly the same naming scheme. I agree it's confusing, and I'd be happy to change it for both of them.

**Comment by ehewins** [comment]
> General comments on the wedge slicers (Windows GUI):
> 
> 1. Regarding the x-axis range:
>    -->The phi slicer has wrong x-axis labels (might there be a pi too much? Its 3.14 "deg" where it should be 0deg
> 2. Regarding the x-axis fonds:
>    -->x-Axis fonds are overlapping for both Q and phi slicers (depending on the axis range), whereas for other kind of slicers there is no problem.

I haven't been able to reproduce either of these issues on my Linux system. The one which confuses me most is the first one, as that definitely shouldn't be OS dependent. What combination of branches were you using? `sasview/main` + `sasdata/master` gives me zero when it should do, and my other branches `sasview/sasview-counterpart-to-sasdata-manipulations-rewrite` + `sasdata/46-manipulations-dot-py-rewrite` do too.

As for the second issue, I've looked through the code to see if there's some kind of adjustment to the tick labels I'd forgotten to copy over to WedgeSlicer from the other slicers, but I can't see anything which might cause WedgeSlicer to behave differently. Anyone else mind giving me a hand with this one?

File: src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py (hunk 2 above)
Comment (ehewins) [reply src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py:238]: On a Linux host I am getting error messages for very small delta_phi angles, either "ValueError: Average Error: No points inside ROI to average..." or "Sector angles too close. Please adjust." depending on where I've placed the slicer. Now that I'm no longer employed by STFC/RAL/ISIS, I don't have access to a windows development machine. Is anyone else able to help with this?

Comment (astellhorn) [reply src/sas/qtgui/Plotting/Slicers/WedgeSlicer.py:238]: I have tried in the main branch (Windows GUI) as well as in the #2570 branch (Windows GUI) 
--> for varying phi/delta_phi: yes, in both it completely depends which phi and which delta_phi together with which nbins I choose. The higher nbins, the smaller delta_phi can be (also <1 and the plot still works)
--> for varying r_min: I can choose at as small as I want (eg 0.00000001) and no error message pops up.
We can hopefully speak about this later in the code camp.

**Comment by astellhorn** [comment]
Ok, I think that has been fixed. The main branch does not give this error, only the #2570 branch (Windows) was giving me weird x-axes labels. However, the second issue with overlapping labels still exists.

**Pull request merged**